test/distributed/elastic/test_control_plane.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+# Owner(s): ["oncall: distributed"]
+
+import os
+import pickle
+import socket
+import tempfile
+from contextlib import contextmanager
+
+import requests
+from requests.adapters import HTTPAdapter
+
+from torch.distributed.elastic.control_plane import main, TORCH_WORKER_SERVER_SOCKET
+from torch.testing._internal.common_utils import requires_cuda, run_tests, TestCase
+from urllib3.connection import HTTPConnection
+from urllib3.connectionpool import HTTPConnectionPool
+
+
+class UnixHTTPConnection(HTTPConnection):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__("localhost")
+
+        self.socket_path = socket_path
+
+    def connect(self) -> None:
+        self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self.sock.connect(self.socket_path)
+
+
+class UnixHTTPConnectionPool(HTTPConnectionPool):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__("localhost")
+
+        self.socket_path = socket_path
+
+    def _new_conn(self):
+        return UnixHTTPConnection(self.socket_path)
+
+
+class UnixHTTPAdapter(HTTPAdapter):
+    def __init__(self, socket_path: str) -> None:
+        super().__init__()
+
+        self.socket_path = socket_path
+
+    def get_connection(self, url, proxies=None):
+        return UnixHTTPConnectionPool(self.socket_path)
+
+    def request_url(self, request, proxies):
+        print(request.path_url)
+        return request.path_url
+
+
+@contextmanager
+def local_worker_server() -> None:
+    with tempfile.TemporaryDirectory() as tmpdir:
+        socket_path = os.path.join(tmpdir, "socket.sock")
+        os.environ[TORCH_WORKER_SERVER_SOCKET] = socket_path
+
+        print("starting")
+        with main():
```

Comment:
can we have a more descriptive name, given that is from another module?

Reply:
I was thinking about following the hydra pattern here: https://hydra.cc/docs/intro/

In practice this should be something like `@worker_server.main`.